### PR TITLE
Update support for Snap 1.0

### PR DESCRIPTION
**Pull request by sigrlami**
Since Snap 1.0 is coming very soon, we need to have code that will be shipped in timely manner along with Snap release. This pull request include:
- update dependencies ( change versions, remove deprecated packages)
- update code with newly added dependencies
- add `stack.yaml` for stack users ( able to quickly get upstream snap 1.0 and relevant packages without hassle)

### CODE REVIEW

File: src/Snap/Snaplet/#SqliteSimple.hs# [hunk 1]
```diff
@@ -0,0 +1,227 @@
+{-# LANGUAGE OverloadedStrings #-}
```
Comment (nurpax) [inline src/Snap/Snaplet/#SqliteSimple.hs#]: did you accidentally add an automatic backup file in your commit, or what is this #SqliteSimple.hs# file? (you have edits to this file below in the patch)

File: src/Snap/Snaplet/SqliteSimple.hs [hunk 2]
```diff
@@ -135,8 +136,8 @@ instance HasSqlite (Handler b Sqlite) where
 -- > d <- nestSnaplet "db" db sqliteInit
 -- > count <- liftIO $ runReaderT (execute "INSERT ..." params) d
 instance (MonadBase IO m) => HasSqlite (ReaderT (Snaplet Sqlite) m) where
-    getSqliteState = asks (\sqlsnaplet -> sqlsnaplet ^# snapletValue)
-
+    getSqliteState = asks (^# snapletValue) -- asks (\sqlsnaplet -> sqlsnaplet ^. snapletValue)
+
```
Comment (nurpax) [inline src/Snap/Snaplet/SqliteSimple.hs]: I prefer no trailing whitespace in this code, so if you need to edit this patch might clean that up.  No big deal if you don't, though.

File: src/Snap/Snaplet/#SqliteSimple.hs# (hunk 1 above)
Comment (nurpax) [inline src/Snap/Snaplet/#SqliteSimple.hs#]: ahh, I see that you removed this in a later commit.  Maybe use interactive rebase to clean up accidental add + delete from history?

Comment (sigrlami) [inline src/Snap/Snaplet/#SqliteSimple.hs#]: yes, sorry for that. Do you want me to prepare another PR?

File: src/Snap/Snaplet/SqliteSimple.hs (hunk 2 above)
Comment (sigrlami) [inline src/Snap/Snaplet/SqliteSimple.hs]: Ok, I'll remove them.

File: snaplet-sqlite-simple.cabal [hunk 3]
```diff
@@ -1,5 +1,5 @@
 name:           snaplet-sqlite-simple
-version:        0.5
+version:        0.5.0.0
```
Comment (nurpax) [inline snaplet-sqlite-simple.cabal]: might as well bump to 0.6.0.0, this version probably drops ghc 7.6 support or at least increases dep requirement so much that we shouldn't call this version 0.5.0.0 (which would be equal to 0.5.*, ie. same as the previous release)

File: src/Snap/Snaplet/#SqliteSimple.hs# (hunk 1 above)
Comment (sigrlami) [inline src/Snap/Snaplet/#SqliteSimple.hs#]: I'll rebase then

**Comment by nurpax** [comment]
This is excellent!  Thanks you @sigrlami so much for doing this.  I added a few comments for clean up ideas.

Also, the travis build fails, but I think that's due to the coverage tracking plugin hpc-coveralls that I use.  I should really just drop it from the travis configuration.

File: src/Snap/Snaplet/#SqliteSimple.hs# (hunk 1 above)
Comment (nurpax) [inline src/Snap/Snaplet/#SqliteSimple.hs#]: No need for a separate PR.  You should be able to use "git rebase -i", clean things up as you desire, and then "git push -f" the same topic into this PR.  

If you've never done that, might want to be careful with doing that.

**Comment by sigrlami** [comment]
@nurpax rebased PR with your suggestions, if you need something else let me know.

**Comment by nurpax** [comment]
What I meant by interactive rebase was that you should squash the commits so that the addition of the temp file disappears from the history.  Maybe I'll just fix these myself when I merge this and make a release.  Which should be hopefully today.

**Comment by sigrlami** [comment]
@nurpax rebased PR again, sorry for hassle, just want to make everything right, in case I'll send new ones, I want it to be in a good shape.

**Comment by Feyrn** [comment]
What still needs to be done for this to be merged?

**Comment by sigrlami** [comment]
@Feyrn everything is ready @nurpax just need to merge it,  if you use stack you can point to my fork https://github.com/sigrlami/snaplet-sqlite-simple/tree/port-1.0?files=1

**Comment by nurpax** [comment]
@sigrlami @Feyrn I haven't worked on this in a while.  I'll merge and release this on Hackage.  Sorry for the delay. :(

**Comment by sigrlami** [comment]
@nurpax would you like me to be back up maintainer? in a way described here https://ro-che.info/articles/2014-02-08-my-haskell-will

**Comment by nurpax** [comment]
@sigrlami Sure!  I'm glad you put that forward.  Do you wanna take a crack at this release? :)

**Comment by nurpax** [comment]
I guess this will have to be merged into master instead of port-1.0.  I don't think we need to care for older snap versions anymore, so might as well be in master.

**Comment by sigrlami** [comment]
@nurpax we can merge it into master, set new version and publish on hackage. Those who works with older snap version will use specific version from hackage

**Comment by nurpax** [comment]
@sigrlami I noticed that the example project fails to build.  Did you by any chance ever look at that?

It's easy to add it to the build by adding

```
- example/
```

under the packages section in stack.yaml.

**Comment by sigrlami** [comment]
@nurpax no, I didn't. Thanks for noticing. I planned to work on this tonight.

**Comment by Feyrn** [comment]
@sigrlami I ended up making a fork of your branch for my project which is using LTS-7.1.
This has the advantage of not needing the stack git sources since LTS-7.1 has snap 1.0.

The fork is here: https://github.com/Feyrn/snaplet-sqlite-simple/tree/port-1.0 if you want to check it out.

**Comment by sigrlami** [comment]
@Feyrn I just updated project; not sure how manage to build example as is. I needed to update several functions in code. It will be soon on hackage and stackage, but for short-term fork is good enough

**Comment by nurpax** [comment]
I have a version that fixes the example project too now.  I'll push to master and release on hackage soonish.

**Comment by sigrlami** [comment]
I tried to push last week to hackage, but there was something wrong with auth that day. Didn't have time this week yet.

**Comment by nurpax** [comment]
Oh, I thought nothing had happened here.  I don't see any updates in the master branch here.  

I upgraded to lts-7.3 locally.  That might be worthwhile before pushing to hackage.

**Comment by nurpax** [comment]
But lts-7.1 based version is hackage is good too. :)

I didn't have to specify a git package location at all.  I just added

extra-deps:
  - snap-loader-static-1.0.0.0
  - heist-1.0.0.0
  - map-syntax-0.2.0.1
  - snap-1.0.0.1
